rest/src/main/java/com/ericsson/bss/cassandra/ecchronos/rest/OnDemandRepairSchedulerREST.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecchronos.rest;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.ericsson.bss.cassandra.ecchronos.core.repair.types.CompleteRepairJob;
+import com.ericsson.bss.cassandra.ecchronos.core.repair.types.ScheduledRepairJob;
+
+public interface OnDemandRepairSchedulerREST
+{
+    /**
+     * Get details of a specific table repair job.
+     *
+     * @param keyspace
+     *            The keyspace of the table
+     * @param table
+     *            The table to get details of
+     * @return A JSON representation of {@link CompleteRepairJob}
+     */
+    @GET
+    @Path("/get/{keyspace}/{table}")
+    @Produces(MediaType.APPLICATION_JSON)
+    String get(@PathParam("keyspace") String keyspace,
+            @PathParam("table") String table);
+
+    /**
+     * Get a list of the status of all repair jobs.
+     *
+     * @return A list of JSON representations of {@link ScheduledRepairJob}
+     */
+    @GET
+    @Path("/list")
+    @Produces(MediaType.APPLICATION_JSON)
+    String list();
+
+    /**
+     * Get a list of the status of all repair jobs for a specific keyspace.
+     *
+     * @param keyspace
+     *            The keyspace to list
+     * @return A list of JSON representations of {@link ScheduledRepairJob}
+     */
+    @GET
+    @Path("/list/{keyspace}")
+    @Produces(MediaType.APPLICATION_JSON)
+    String listKeyspace(@PathParam("keyspace") String keyspace);
+
+    @GET
```

Comment:
We can always switch it to POST if needed in the future. I don't think we should make this change if we don't necessarily need it at the moment.

Reply:
I think it is wrong to start a repair job with GET. I also don't think it follows REST. How is it possible GET a resource that was never created (POST/PUT). 

Changing the method in the future would mean that we'd break backwards compatibility?

Reply:
I understand where you are coming from. I had the exact same feeling at one point and I still don't feel good about GET.

You correctly pointed out the fallacies of using GET but you fail to mention that PUT/POST are not great in this scenario either. We aren't actually putting/posting any data so PUT and POST are also not 'right'. Unfortunately REST isn't really designed to be used in such context.

The only reason I am leaning towards GET is because we are currently not actually sending any data. (PUT and POST expect a payload to be "posted or put")

As emolsson correctly pointed out, this will feel better when we do have the repair configuration properties to send in. There is an argument to make it POST and sending an empty payload for the sake of compatibility with the future version. I might be swayed to change it to POST/PUT for that reason, although only if it doesn't require any strange workarounds in the integration tests. I will have a look to see how feasible that is.

Reply:
Got your point. But it is actually fine to send POST without a payload. https://stackoverflow.com/questions/4191593/is-it-considered-bad-practice-to-perform-http-post-without-entity-body

As manuel-aldana said:
> POST is completely OK. In difference of GET with POST you are changing the state of the system (most likely your trigger is "doing" something and changing data).

That's exactly what we're doing as well. As you can see I still heavily favor using POST here but I also realize I'm not really in the position of making that decision. Up to you.

Reply:
Maybe not an issue in practice, but when considering caching - POST is also preferable when "triggering an action" via REST...
"Using a POST instead of a GET is perfectly reasonable, since it also instructs the server (and gateways along the way) not to return a cached response"

Reply:
Valid points. Consider me convinced :) I will look into changing this.